### ISSUE TITLE
Conversion error: "the estimator object does not specify the number of outputs" 

### ISSUE THREAD

**Issue by Dopidotcsv** [jpmml/sklearn2pmml#357]
Hi, I'm not sure if I can create a Issue with this kind of question, but here it goes:

I've my sklearn pipeline fitted and converte to PMMLPipeline with make_pmml_pipeline(), but tryng to export it with sklearn2pmml() gets this error:

```b
Standard output is empty
Standard error:
Picked up JAVA_TOOL_OPTIONS: -Dhttp.proxyHost=proxy-gdpshs-p.we1.azure.aztec.cloud.allianz -Dhttp.proxyPort=80 -Dhttp.nonProxyHosts=10.*|localhost|127.0.0.1|*.allianz|*.cluster.local|*.svc.cluster.local|*.svc|169.254.169.254|44.138.224.* -Dhttps.proxyHost=proxy-gdpshs-p.we1.azure.aztec.cloud.allianz -Dhttps.proxyPort=80 -Dhttps.nonProxyHosts=10.*|localhost|127.0.0.1|*.allianz|*.cluster.local|*.svc.cluster.local|*.svc|169.254.169.254|44.138.224.*
Exception in thread "main" java.lang.IllegalArgumentException: The estimator object of the final step (Python class lightgbm.sklearn.LGBMClassifier) does not specify the number of outputs
	at sklearn2pmml.pipeline.PMMLPipeline.initTargetFields(PMMLPipeline.java:564)
	at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:132)
	at com.sklearn2pmml.Main.run(Main.java:91)
	at com.sklearn2pmml.Main.main(Main.java:66)
``` 
I've been looking around to find similar error with nº of outputs not specified with LGBMCLassifier but didn't found anything.

To clarify: 
I'm getting the pipeline from a  db, it is already fitted with the proper data and I'm working only with the pipeline and trying to export it as pmml. 
Thanks

**Comment by vruusmann** [comment]
> Exception in thread "main" java.lang.IllegalArgumentException: The estimator object of the final step (Python class lightgbm.sklearn.LGBMClassifier) does not specify the number of outputs

The SkLearn2PMML package has multi-label (aka multi-target) model support since the 0.86.0 version (released in early September, 2022).

Earlier package versions were always assuming a single label. The latest version(s) require you to specify the number of labels explicitly - even if it's a single target.

Looks like this change has broken many workflows, and the JPMML-SkLearn library should include a "heuristic" that if the `PMMLPipeline.target_fields` attribute is not set, then it should be soft-replaced with a single label called `y`.

It is my position that when you get this exception, it means that you're using the SkLearn2PMML package conceptually wrong - you workflow is lacking very fundamental information (the number and name of labels). But if that's okay by you, then so be it.

**Comment by vruusmann** [comment]
> I've my sklearn pipeline fitted and converte to PMMLPipeline with make_pmml_pipeline()

The `sklearn2pmml.make_pmml_pipeline()` function takes `active_fields` and `target_fields` arguments. You should be setting the latter argument as a temporary workaround:

```python
my_prefitted_model = pickle.loads(..)

pmml_pipeline = make_pmml_pipeline(my_prefitted_model, target_fields = ["y"])
sklearn2pmml(pmml_pipeline, "pipeline.pmml")
```

**Comment by vruusmann** [comment]
The converter is actually looking for the `<estimator>.n_outputs_` attribute, before bailing out with an error.

This attribute should be always present in newer Scikit-Learn versions. So, perhaps upgrading the Scikit-Learn package version may solve the problem in some cases.

Sure, this user is working with archived Scikit-Learn pipeline objects, so a Scikit-Learn package version upgrade won't help. One can always perform a manual pipeline object "upgrade", by setting the `n_outputs_` attribute explicitly.

```python
my_prefitted_pipeline = pickle.loads(..)

# THIS!
my_prefitted_pipeline._final_estimator.n_outputs_ = 1

pmml_pipeline = make_pmml_pipeline(my_prefitted_pipeline)
sklearn2pmml(pmml_pipeline, "pipeline.pmml")
```
